src/Services/E-Games.Services/E-Games.Services/GameService.cs

```diff
@@ -0,0 +1,46 @@
+﻿using E_Games.Common.DTOs;
+using E_Games.Data.Data;
+using E_Games.Web.Exceptions;
+using Microsoft.EntityFrameworkCore;
+
+namespace E_Games.Services.E_Games.Services
+{
+    public class GameService : IGameService
+    {
+        private readonly ApplicationDbContext _context;
+
+        public GameService(ApplicationDbContext context) => _context = context;
+
+        public async Task<List<PlatformPopularityDto>> GetTopPlatformsAsync() => await _context.Products
+                .GroupBy(p => p.Platform)
+                .Select(group => new PlatformPopularityDto
+                {
+                    PlatformName = group.Key.ToString(),
+                    Count = group.Count()
+                })
+                .OrderByDescending(dto => dto.Count)
+                .Take(3)
+                .ToListAsync();
+
+        public async Task<List<SearchGameDto>> SearchGamesAsync(string term, int limit, int offset)
```

Comment:
Generally it is fine to use primitives, I don't require no changes here, but pay attention. As you're adding more arguments, it makes sense to pass dto classes instead